.github/workflows/scripts/sync-deps.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+#
+#
+
+set -e
+
+RANCHER_REPO_DIR=$1
+# File to write the changes to
+CHANGES_FILE=${2:-/dev/null}
+
+DEPS_TO_SYNC="
+  github.com/rancher/remotedialer
+  github.com/rancher/dynamiclistener
+  github.com/rancher/wrangler/v3
+"
+
+if [ -z "$RANCHER_REPO_DIR" ]; then
+  usage
+  exit 1
+fi
+
+usage() {
+  echo "$0 <path to rancher repository> [<path to write dependency changes to>]"
+}
+
+update_dep() {
+  module=$1
+  old_version=$2
+  new_version=$3
+
+  echo "Version mismatch for $module (rancher=$new_version, rdp=$old_version) detected"
+  go mod edit -require="$module@$new_version"
+  printf '**%s**\n`%s` => `%s`\n' "$module" "$old_version" "$new_version" >> "$CHANGES_FILE"
+}
+
+rancher_deps=$(cd "$RANCHER_REPO_DIR" && go mod graph)
+rdp_deps=$(go mod graph)
```

Comment:
Prob better toc all this "remotedialerproxy_deps". It's only set once and used once so a longer name fits with `rancher_deps` and isn't onerous to read